### PR TITLE
LPS-174767 | Test Automation | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# Move the Segments Page to the Control Panel

Hi! :)
This pr is corresponding to these tickets: 
- [LPS-167023](https://issues.liferay.com/browse/LPS-167023)
- [LPS-174767](https://issues.liferay.com/browse/LPS-174767)

| **Test File** | **Test Name** |
| --- | --- |
|  SegmentationCreateSegment | SegmentPortletNotPresentInProductMenu |
|  SegmentationCreateSegment | SegmentInGlobalSiteByDefault |
|  SegmentationCreateSegment | SiteNameInScopeColumn |
|  SegmentationAssigningRoles | SegmentsPermissionsAllocatedInControlPanel |
| SegmentationAssigningRoles | SegmentsPermissionsNotAllocatedInSiteAdministration |
|  SegmentationAssigningRoles| SegmentsPermissionsHaveAccessToControlPanel |
|  SegmentationAssigningRoles| UserWithPermissionsCanAccessSegmentsPortlet |
|  SegmentationAssigningRoles| UserWithSegmentPermissionsWithoutControlPanelCannotAccess |
|  SegmentationAssigningRoles| UserWithoutPermissionsCannotAccessSegmentsPortlet |

### CODE REVIEW

**Comment by rafaellajordao** [comment]
ci:test:sf

**Comment by rafaellajordao** [comment]
ci:test:tango

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase [hunk 1]
```diff
@@ -729,4 +729,259 @@ definition {
...
+				key_permissionNavHeader = "Site and Asset Library Administration",
+				key_permissionNavItem = "Segments",
+				locator1 = "RolesPermissionsNavigation#PERMISSIONS_NAVIGATION");
+		}
+	}
+
+	@description = "LPS-167023 Given a user without Segments permissions
+	When the user clicks on Applications Menu > Control Panel
+	Then the Segments portlet is not present and the user cannot access to the Segments page"
+	@priority = "5"
```
Comment (rafaellajordao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase:829]: Hey @CarlosBrichete , I have a doubt! When the test has the Feature Flag property, should I set the Acceptance property?

Comment (CarlosBrichete) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase:829]: Yes, you can set both properties

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase [hunk 2]
```diff
@@ -729,4 +729,259 @@ definition {
 		}
 	}
 
+	@description = "LPS-167023 Given the Roles portlet
+	When the user tries to add a new role
+	And the user wants to add segments permissions
+	Then the segments permissions are allocated in Control Panel > Users"
+	@priority = "4"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase:736]: Add the property portal.acceptance

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase [hunk 3]
```diff
@@ -729,4 +729,259 @@ definition {
...
+				key_permissionNavHeader = "Control Panel",
+				key_permissionNavItem = "Segments",
+				locator1 = "RolesPermissionsNavigation#PERMISSIONS_NAVIGATION");
+		}
+	}
+
+	@description = "LPS-167023 Given the user creates a new role
+	When the user clicks on Control Panel > Users > Segments
+	Then in General Permissions, the check Access in Control Panel permission is displayed"
+	@priority = "4"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase:765]: Add the property portal.acceptance

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase (hunk 1 above)
Comment (CarlosBrichete) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase:829]: Add the property portal.acceptance

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase [hunk 4]
```diff
@@ -729,4 +729,259 @@ definition {
...
+				locator1 = "Portlet#ERROR",
+				value1 = "You do not have the roles required to access this portlet.");
+		}
+	}
+
+	@description = "LPS-167023 Given a user with Segments permissions
+	And with View Control Panel Menu permission
+	When the user clicks on Applications Menu > Control Panel > Segment
+	Then the user can access to the Segments page"
+	@priority = "5"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase:867]: Add the property portal.acceptance

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase [hunk 5]
```diff
@@ -729,4 +729,259 @@ definition {
...
+				requireReset = "false",
+				userEmailAddress = "userea@liferay.com");
+		}
+
+		task ("Create a new role through Roles Admin Portlet") {
+			Role.openRolesAdmin();
+
+			Navigator.gotoNavTab(navTab = "Regular Roles");
+
+			Role.add(roleTitle = "UserWithPermissionsCanAccessSegmentsPortlet Role");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase:889]: Add role using JSONRole.addRegularRole

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase [hunk 6]
```diff
@@ -729,4 +729,259 @@ definition {
...
+			Role.editPermissionCP(
+				define = "true",
+				permissionNavHeader = "Users",
+				permissionNavItem = "Segments",
+				permissionSheetSubtitle = "Application Permissions",
+				permissionValues = "Access in Control Panel");
+
+			Role.viewPermissionCP(
+				permissionDefinitionValue = "View Control Panel Menu",
+				permissionNavigationValue = "Portal");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase:909]: If possible, add permissions using JSON call Permissions.definePermissionViaJSONAPI

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase [hunk 7]
```diff
@@ -729,4 +729,259 @@ definition {
...
+				locator1 = "Message#EMPTY_INFO",
+				value1 = "There are no segments.");
+		}
+	}
+
+	@description = "LPS-167023 Given a user with Segments permissions
+	And without View Control Panel Menu permission
+	When the user clicks on Applications Menu > Control Panel
+	Then the Segments portlet is not present and the user cannot access to the Segments page"
+	@priority = "5"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase:938]: Add the property portal.acceptance

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase [hunk 8]
```diff
@@ -729,4 +729,259 @@ definition {
...
+				requireReset = "false",
+				userEmailAddress = "userea@liferay.com");
+		}
+
+		task ("Create a new role through Roles Admin Portlet") {
+			Role.openRolesAdmin();
+
+			Navigator.gotoNavTab(navTab = "Regular Roles");
+
+			Role.add(roleTitle = "UserWithSegmentPermissionsWithoutControlPanelCannotAccess Role");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase:960]: Add role using JSONRole.addRegularRole

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase [hunk 9]
```diff
@@ -729,4 +729,259 @@ definition {
...
+		}
+
+		task ("Define permissions without View Control Panel Menu permission") {
+			Role.editPermissionCP(
+				define = "true",
+				permissionNavHeader = "Users",
+				permissionNavItem = "Segments",
+				permissionSheetSubtitle = "Segments",
+				permissionValues = "Permissions");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationAssigningRoles.testcase:969]: If possible, add permissions using JSON call Permissions.definePermissionViaJSONAPI

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationCreateSegment.testcase [hunk 10]
```diff
@@ -3040,6 +3040,78 @@ definition {
 		}
 	}
 
+	@description = "LPS-167023 Given the Segments page
+	When the user adds a new segment
+	Then the segment is created in the Global Site"
+	@priority = "4"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationCreateSegment.testcase:3046]: Add the property portal.acceptance

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationCreateSegment.testcase [hunk 11]
```diff
@@ -3040,6 +3040,78 @@ definition {
...
+				locator1 = "Segmentation#SCOPE_CARD_SITE",
+				siteName = "Global",
+				value1 = "Global");
+		}
+	}
+
+	@description = "LPS-167023 Given the Product Menu
+	When the user clicks on People
+	Then the Segments portlet is not present"
+	@priority = "4"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/SegmentationCreateSegment.testcase:3067]: Add the property portal.acceptance

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]

**Comment by rafaellajordao** [comment]
ci:test:tango

**Comment by rafaellajordao** [comment]
ci:test:sf

**Review by CarlosBrichete — APPROVED** [review]

**Comment by CarlosBrichete** [comment]
ci:forward

**Pull request closed without merging**